app/src/main/java/com/arctouch/publictransportation/fragment/StopListFragment.java

```diff
@@ -0,0 +1,185 @@
+package com.arctouch.publictransportation.fragment;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.android.volley.RequestQueue;
+import com.arctouch.publictransportation.R;
+import com.arctouch.publictransportation.activity.DepartureListActivity;
+import com.arctouch.publictransportation.domain.Route;
+import com.arctouch.publictransportation.domain.Stop;
+import com.arctouch.publictransportation.domain.generic.params.RouteIdParam;
+import com.arctouch.publictransportation.domain.generic.returns.StopsReturn;
+import com.arctouch.publictransportation.interfaces.VolleyListener;
+import com.arctouch.publictransportation.volley.VolleySingleton;
+import com.arctouch.publictransportation.volley.endpoints.StopsVolley;
+
+import java.util.List;
+
+public class StopListFragment extends Fragment implements VolleyListener<StopsReturn> {
+    public static final String ROUTE = "route";
+    public static final String SHOW_ROUTE_DETAILS = "show_route_details";
+
+    private View mRootView;
+    private View mRecyclerView;
+    private ProgressBar mProgressBar;
+    private TextView mEmptyList;
+    private Button mSeeDepartures;
+
+    private boolean mShowRouteDetails;
+
+    private Route mRoute;
+
+    public StopListFragment() {
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (getArguments().containsKey(ROUTE)) {
+            mShowRouteDetails = getArguments().containsKey(SHOW_ROUTE_DETAILS) &&
+                    getArguments().getBoolean(SHOW_ROUTE_DETAILS);
+
+            mRoute = (Route) getArguments().getSerializable((ROUTE));
+        } else {
+            throw new IllegalArgumentException("StopListFragment needs an argument type Route.");
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        this.mRootView = inflater.inflate(R.layout.layout_stop_list, container, false);
+
+        this.mRecyclerView = mRootView.findViewById(R.id.stopList);
+        this.mProgressBar = (ProgressBar) mRootView.findViewById(R.id.progressBar);
+        this.mEmptyList = (TextView) mRootView.findViewById(R.id.emptyList);
+        this.mSeeDepartures = (Button) mRootView.findViewById(R.id.seeDepartures);
+
+        TextView routeName = (TextView) mRootView.findViewById(R.id.routeName);
+        View seeDepartures = (View) mRootView.findViewById(R.id.seeDepartures);
```

Comment:
Three things: One, casting to _View_ is unnecessary, because _findViewById_ already returns one;
Second, it's a good practice to always cast to the actual component type (in this case, a _Button_);
Third, this component is already being "found" two lines above (in mSeeDepartures), so you should remove this line and use that instead